File: src/game/tasks/GameTickerObserver.java
package game.tasks;

import game.events.models.Observer2;

public abstract class GameTickerObserver implements Observer2 {
    private static final float DEFAULT_DELAY_OBSERVER = 30;
    protected long lastUpdate = 0L;

    @Override
    public void update(Object arg) {
        lastUpdate = System.currentTimeMillis();
    }

    public long getLastUpdate() {
        return lastUpdate;
    }

    protected float getDelayObserverUpdate() {
        return DEFAULT_DELAY_OBSERVER;
    }
}

File: src/game/tasks/GameTickerObserver.kt
package game.tasks

import game.events.models.Observer2

abstract class GameTickerObserver : Observer2 {
    var lastUpdate = 0L
        protected set

    override fun update(arg: Any?) {
        lastUpdate = System.currentTimeMillis()
    }

    open val delayObserverUpdate: Float
        get() {
            return 30f
        }
}